src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateAttributeDefinitionActionUtils.java

```diff
@@ -158,15 +159,44 @@
                 final AttributeDefinitionDraft matchingNewAttributeDefinitionDraft =
                     newAttributeDefinitionDraftsNameMap.get(oldAttributeDefinitionName);
                 return ofNullable(matchingNewAttributeDefinitionDraft)
-                    .map(attributeDefinitionDraft ->
-                        buildActions(oldAttributeDefinition, attributeDefinitionDraft)
-                    )
+                    .map(attributeDefinitionDraft -> {
+                        // attribute type is required so if null we let commercetools to throw exception
+                        if (attributeDefinitionDraft.getAttributeType() != null) {
+                            if (haveSameAttributeType(oldAttributeDefinition, attributeDefinitionDraft)) {
+                                return buildActions(oldAttributeDefinition, attributeDefinitionDraft);
+                            } else {
+                                return Arrays.asList(
+                                    RemoveAttributeDefinition.of(oldAttributeDefinitionName),
+                                    AddAttributeDefinition
+                                        .of(AttributeDefinitionCustomBuilder.of(attributeDefinitionDraft))
+                                );
+                            }
+                        } else {
+                            return new ArrayList<UpdateAction<ProductType>>();
```

Comment:
but I don't see you doing as the comment actually says, so if the attributeType on the draft is null, no actions will be built and also the user won't get any warning/error?

Why not actually warn the user about the null type?

Reply:
Attribute type is required by commercetools, so if I'm not mistaken, in one conversation we had (and previously discussed with Roman), we decided to not add the validations in the library that are documented in commercetools  (like required values), and let commercetools to throw the error.

Did I get it wrong when we discussed it?

Reply:
- Hmm, but the thing is I don't see how CT will throw an exception this case. Because this method will return an empty list so no actual update action will be issued to CT. Or do you mean only in case if there is an update action from the calling method (`ProductTypeUpdateAttributeDefinitionActionUtils#buildUpdateActions`) then CT will throw an exception?

- Also, Please remove the `throws DuplicateKeyException` declaration on the method signature ;)

Reply:
> do you mean only in case if there is an update action from the calling method (ProductTypeUpdateAttributeDefinitionActionUtils#buildUpdateActions) then CT will throw an exception?

Yes, I meant that. If there is any update action CT will throw an error no matter the update actions

Reply:
> Also, Please remove the throws DuplicateKeyException declaration on the method signature ;)

9f0255dafd8b7fd1a10703e21913f6182ce3cb34